core/include/detray/propagator/navigator.hpp

```diff
@@ -398,7 +398,8 @@ class navigator {
         DETRAY_HOST_DEVICE inline auto is_on_object(
             const intersection_type &candidate,
             const navigation::config &cfg) const -> bool {
-            return (candidate.path < math_ns::abs(cfg.on_surface_tolerance));
+            return (math_ns::abs(candidate.path) <
+                    math_ns::abs(cfg.on_surface_tolerance));
```

Comment:
You are right, I added the abs function on the wrong path, but we don't need to call it on the on surface tolerance again (it is always positive):
```
```suggestion
            return (math_ns::abs(candidate.path) < cfg.on_surface_tolerance);
```

Reply:
OK Fixed